plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java

```diff
@@ -0,0 +1,421 @@
+                .hasMessage("line 1:21: Table function system.query not registered");
+        assertFalse(getQueryRunner().tableExists(getSession(), "numbers"));
+    }
+
+    @Override
+    public void testNativeQueryInsertStatementTableDoesNotExist()
+    {
+        // table function disabled for Ignite, because it doesn't provide ResultSetMetaData, so the result relation type cannot be determined
+        assertFalse(getQueryRunner().tableExists(getSession(), "non_existent_table"));
+        assertThatThrownBy(() -> query("SELECT * FROM TABLE(system.query(query => 'INSERT INTO non_existent_table VALUES (1)'))"))
+                .hasMessage("line 1:21: Table function system.query not registered");
+    }
+
+    @Override
+    public void testNativeQueryInsertStatementTableExists()
+    {
+        // table function disabled for Ignite, because it doesn't provide ResultSetMetaData, so the result relation type cannot be determined
+        try (TestTable testTable = simpleTable()) {
+            assertThatThrownBy(() -> query(format("SELECT * FROM TABLE(system.query(query => 'INSERT INTO %s VALUES (3, 4)'))", testTable.getName())))
+                    .hasMessage("line 1:21: Table function system.query not registered");
+            assertQuery("SELECT * FROM " + testTable.getName(), "VALUES (1, 1), (2, 2)");
+        }
+    }
+
+    @Override
+    protected TestTable simpleTable()
+    {
+        return new TestTable(onRemoteDatabase(), format("%s.simple_table", getSession().getSchema().orElseThrow()), "(col BIGINT, id bigint primary key)", ImmutableList.of("1, 1", "2, 2"));
+    }
+
+    @Override
+    public void testNativeQueryIncorrectSyntax()
+    {
+        // table function disabled for Ignite, because it doesn't provide ResultSetMetaData, so the result relation type cannot be determined
+        assertThatThrownBy(() -> query("SELECT * FROM TABLE(system.query(query => 'some wrong syntax'))"))
+                .hasMessage("line 1:21: Table function system.query not registered");
+    }
+
+    @Override
+    public void testCharVarcharComparison()
+    {
+        // Ignite will map char to varchar, skip
+        throw new SkipException("Ignite map char to varchar, skip test");
+    }
+
+    @Override
+    protected String errorMessageForInsertIntoNotNullColumn(String columnName)
+    {
+        return format("Failed to insert data: Null value is not allowed for column '%s'", columnName.toUpperCase(Locale.ENGLISH));
+    }
+
+    @Override
+    public void testCharTrailingSpace()
+    {
+        throw new SkipException("Ignite not support char trailing space");
+    }
+
+    @Override
+    protected Optional<DataMappingTestSetup> filterDataMappingSmokeTestData(DataMappingTestSetup dataMappingTestSetup)
+    {
+        switch (dataMappingTestSetup.getTrinoTypeName()) {
+            case "date":
+                return Optional.empty();
```

Comment:
I expect we can pass `DATE '2020-02-12'` & `DATE '9999-12-31'` case. Please add another condition:
```java
            case "date":
                // Ignite doesn't support these days
                if (dataMappingTestSetup.getSampleValueLiteral().equals("DATE '0001-01-01'") || dataMappingTestSetup.getSampleValueLiteral().equals("DATE '1582-10-05'")) {
                    return Optional.empty();
                }
...
```